client/src/commands.ts

```diff
@@ -164,19 +168,21 @@ export function startLanguageServer(
     env["NO_COLOR"] = "1";
     env["DENO_V8_FLAGS"] = getV8Flags();
 
+    const shell = process.platform === "win32" &&
+      /\.([Cc][Mm][Dd]|[Bb][Aa][Tt])$/.test(command);
     const serverOptions: ServerOptions = {
       run: {
         command,
         args: ["lsp"],
-        options: { env },
-      },
+        options: { env, shell },
+      } as Executable,
       debug: {
         command,
         // disabled for now, as this gets super chatty during development
         // args: ["lsp", "-L", "debug"],
         args: ["lsp"],
-        options: { env },
-      },
+        options: { env, shell },
+      } as Executable,
```

Comment:
Why are the casts necessary? `as Executable`.

Reply:
It was just for typed intellisense so I knew what data type was expected. Without the cast, the structs weren't typed appropriately.

If necessary, I can revert it or use narrowing and type extraction to extract the correct type from the ServerOptions union.